.github/api_best_practices.md

```diff
@@ -0,0 +1,198 @@
+# ACS API Guidelines
```

Comment:
Please mention kinds of APIs this document focuses because they are different. REST, gRPC, GraphQL, Kubernetes (CRDs).

Reply:
This document intends to provide guidelines for all APIs, which will come in separate PRs. I don't see the point of enlisting each category here, for that there is a table of contents.

Reply:
My point is in similar direction as Van's comment about URLs. The following section `Naming Guidelines` talks about gRPC and proto naming. How does the reader know that? Well, some amount of experience will allow inferring that context from the content itself, but possessing that context is already high enough bar.

I think, this bar is set higher than necessary because we'd want less experienced engineers to also benefit from reading these guidelines. Anyone can read anything, the question here is that we make it harder to comprehend.

The second, related, part of my point is that the presented naming guidelines make little sense for Kubernetes APIs which are mostly resource objects (nouns) instead of actions (verbs). If someone wants to put guidelines on Kubernetes APIs naming, how does that person know if this document is fit for that?

The last part of my point is that by making it explicit which API kinds will be in the document, you'd set expectations for reviewers about what they'll find in the future iterations. You'd also be able to source info from reviewers experience about important differences if certain guidelines don't apply for all kinds.

My honest fear is that the guidelines are going to default to gRPC and the rest will be an afterthought, at best. It's fine to address gRPC APIs only but let's be explicit about that. The opposite of doing gRPC with few touches about REST, GraphQL and/or Kubernetes would leave the first impression of something thorough but makes bitter taste when applied in practice because the other kinds are substantially different in some aspects to be covered in a manner of deviations from/extensions of gRPC.

Reply:
> The following section Naming Guidelines talks about gRPC and proto naming. How does the reader know that?

Fair point. The individual section title can be expanded like `Method Name` -> `RPC Method Name` and likewise. I will do that. 

I want to mention the PRs will be much smaller chunks, therefore, we should refrain from asking for things that are not the PR, and rather focus on things that are part of the PR.